zerver/openapi/zulip.yaml

```diff
@@ -8261,6 +8266,89 @@ paths:
       responses:
         "200":
           $ref: "#/components/responses/SimpleSuccess"
+  /user_topics:
+    post:
+      operationId: update-user-topic
+      summary: Update visibility policy of a topic
+      tags: ["streams"]
+      description: |
+        This endpoint is used to update the visibility policy of a topic.
+
+        - 0 - INHERIT
+        - 1 - MUTED
+        - 2 - UNMUTED
+
+        The visibility policy, when set to MUTED, mutes a topic;
+        when set to UNMUTED, it unmutes a topic in a muted stream;
+        and INHERIT is used to remove the visibility policy already set.
+
+        This endpoint mutes or unmutes a topic in a stream or unmutes a topic
+        in a muted stream that the current user is subscribed to.
+        MUTED topics are displayed faded in the Zulip UI and are not included
+        in the user's unread count totals. An UNMUTED topic in a muted stream
+        behaves as an unmuted topic in an unmuted stream.
+
+        **Changes**: New in Zulip 7.0 (feature level 170).
+        The `/users/me/subscriptions/muted_topics` endpoint, which used to only
+        mute or unmute a topic, is now deprecated.
+        It exists for backwards compatibility and will be removed once clients
+        have migrated to use this endpoint.
+      x-curl-examples-parameters:
+        oneOf:
+          - type: exclude
+            parameters:
+              enum:
+                - stream_id
+      parameters:
+        - name: stream_id
+          in: query
+          description: |
+            The ID of the stream to access.
+
+            Clients must provide either `stream_name` or `stream_id` as a parameter
+            to this endpoint, but not both.
+          schema:
+            type: integer
+          example: 43
+          required: false
+        - name: stream_name
+          in: query
+          description: |
+            The name of the stream to access.
+
+            Clients must provide either `stream_name` or `stream_id` as a parameter
+            to this endpoint, but not both. Clients should use `stream_id`
+            instead of the `stream_name` parameter when possible.
```

Comment:
Do we need to support `stream_name`? I think we have it on the other endpoint for backwards-compatibility, but I'd prefer to just not offer it here given that it's a new endpoint.

Reply:
Cool, removed `stream_name` -- `stream_id` is now a required parameter.